Automation: Inject RuleManager instead of non existing ModuleHandlerCallback

triller-telekom opened the pull request:
Fixes #5976

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
-     * Bind the {@link ModuleHandlerCallback} service - called from DS.
+     * Bind the {@link RuleEngine} service - called from DS.
      *
      * @param ruleEngine RuleManager service.
```
ruleEngine -> ruleManager

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
 
     /**
-     * Bind the {@link ModuleHandlerCallback} service - called from DS.
+     * Bind the {@link RuleEngine} service - called from DS.
```
RuleEngine -> RuleManager

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -142,13 +146,13 @@ protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegist
-    protected void setRuleEngine(ModuleHandlerCallback ruleEngine) {
-        this.callback = ruleEngine;
+    protected void setRuleManager(RuleManager ruleManager) {
+        this.callback = ruleManager;
```
probably also worth to rename `callback` to `ruleManager`.

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -43,7 +45,7 @@
  * @author Kai Kreuzer - refactored (managed) provider and registry implementation
  */
-@Component(immediate = true)
+@Component(immediate = true, name = "org.eclipse.smarthome.automation.commands")
```
Do we require a specific name here?

triller-telekom replied:
It was like that before in the xml file, which is why I kept it.

triller-telekom replied:
Just tested it: Works without a specific name.

triller-telekom commented:
Comments are addressed

maggu2810 commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -408,8 +412,8 @@ protected String buildCommandUsage(final String syntax, final String description
     public RuleStatus getRuleStatus(String ruleUID) {
-        if (callback != null) {
-            RuleStatusInfo rsi = callback.getStatusInfo(ruleUID);
+        if (ruleManager != null) {
```
The rule manager is a mandatory reference, so it cannot be null if the component is active.

maggu2810 commented on bundles/automation/org.eclipse.smarthome.automation.commands/src/main/java/org/eclipse/smarthome/automation/internal/commands/AutomationCommandsPluggable.java:
```diff
@@ -418,8 +422,8 @@ public RuleStatus getRuleStatus(String ruleUID) {
     public void setEnabled(String uid, boolean isEnabled) {
-        if (callback != null) {
-            callback.setEnabled(uid, isEnabled);
+        if (ruleManager != null) {
```
See above (lifecycle)

triller-telekom commented:
You are right, the `AutomationCommandsPluggable` does not have to be activated immediately. I have removed the immediate activation. I also removed the `null` checks for the mandatory reference.
